src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +10,22 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        valiidateUser(user);
+        return storageDao.add(user);
+    }
+
+    private void valiidateUser(User user) {
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidUserException("Invalid login: " + user.getLogin());
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidUserException("Invalid password: " + user.getPassword());
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
+            throw new InvalidUserException("Invalid age: " + user.getAge());
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserException("User with login already exists" + user.getLogin());
+        }
```

Comment:
1. Please extract magic values to constants with proper naming
2. Please make exception messages hold information to determine what is wrong with login or pass. Message "Invalid login: john" doesn't give any clue to user that he should use at least 6 symbols